src/main/scala/com/mwz/sonar/scala/metadata/model.scala

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (C) 2018-2020  All sonar-scala contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Lesser Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.mwz.sonar.scala.metadata
+
+import cats.data.Chain
+import enumeratum._
+import org.sonar.api.server.rule.RuleParamType
+
+final case class RulesRepository(
+  key: String,
+  name: String,
+  rules: Chain[Rule]
+)
+
+final case class Rule(
+  key: String,
+  name: String,
+  description: String,
+  severity: Severity,
+  template: Boolean,
+  params: Chain[Param]
+)
+
+final case class Param(
+  name: String,
+  typ: ParamType,
+  description: String,
+  default: String
+)
+
+sealed trait ParamType extends EnumEntry
+object ParamType extends Enum[ParamType] with CirceEnum[ParamType] with CatsEnum[ParamType] {
+  case object String extends ParamType
+  case object Text extends ParamType
+  case object Boolean extends ParamType
+  case object Integer extends ParamType
+  case object Float extends ParamType
```

Comment:
AFAIK, **case objects** should also be `final`.
https://nrinaudo.github.io/scala-best-practices/adts/final_case_objects.html
https://github.com/scala/bug/issues/11094

Reply:
nice, I'm glad this is sorted in dotty